presto-main/src/main/java/com/facebook/presto/operator/aggregation/differentialentropy/DifferentialEntropyAggregation.java

```diff
@@ -114,17 +77,12 @@ public static void input(
             @SqlType(StandardTypes.BIGINT) long size,
             @SqlType(StandardTypes.DOUBLE) double sample)
     {
-        DifferentialEntropyStateStrategy strategy = state.getStrategy();
-        if (state.getStrategy() == null) {
-            strategy = new UnweightedReservoirSampleStateStrategy(size);
-            state.setStrategy(strategy);
-        }
-        else {
-            verify(strategy instanceof UnweightedReservoirSampleStateStrategy,
-                    format("Expected UnweightedReservoirSampleStateStrategy, got: %s", strategy.getClass().getSimpleName()));
-        }
-        strategy.validateParameters(size, sample);
-        strategy.add(sample, 1.0);
+        DifferentialEntropyStateStrategy strategy = DifferentialEntropyStateStrategy.getStrategy(
+                state.getStrategy(),
+                size,
+                sample);
+        state.setStrategy(strategy);
+        strategy.add(sample);
```

Comment:
It is now done internally by the strategy base class. Since the point of this diff is to remove duplication between users of the strategy class, I'd like to remove the onus of passing the 1.0 from the callers of the class to the class itself internally.